src/NotificationTarget.php

```diff
@@ -1710,4 +1734,22 @@ public function setAllowResponse($allow_response)
 
         return $this;
     }
+
+    /**
+     * Check if at least one target item is recursive
+     *
+     * @return bool
+     */
+    protected function mustCheckChildrenEntities(): bool
```

Comment:
IMHO, if there are multiple target objects and one of them is NOT recursive, we should not include recipients that have only access to child entities.

Also, maybe the method should be named `canRecipientsFromAddChildEntities` or something like that.

Reply:
But wouldn't some valid users still don't get notifications in this case ?
I don't see how to get a perfect solution without huge breaking changes, the fact that an event can accept multiple items is a huge problem to handle entities correctly

Reply:
Maybe I could target `$this->obj` instead, they its the responsibility of each notification to be triggered correctly ?